interfaces/builtin/dbus_app_test.go

```diff
@@ -0,0 +1,467 @@
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - `)
+	mockSnapYaml = append(mockSnapYaml, long_name...)
+	mockSnapYaml = append(mockSnapYaml, "\n"...)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, Not(IsNil))
+	c.Assert(err, ErrorMatches, "bus name is too long \\(must be <= 255\\)")
+}
+
+func (s *DbusAppInterfaceSuite) TestGetBusNamesNameStartsWithColon(c *C) {
+	var mockSnapYaml = []byte(`name: dbus-app-snap
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - :dbus-app-snap.bar
+`)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, Not(IsNil))
+	c.Assert(err, ErrorMatches, "invalid bus name: \":dbus-app-snap.bar\"")
+}
+
+func (s *DbusAppInterfaceSuite) TestGetBusNamesNameStartsWithNum(c *C) {
+	var mockSnapYaml = []byte(`name: dbus-app-snap
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - 0dbus-app-snap.bar
+`)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, Not(IsNil))
+	c.Assert(err, ErrorMatches, "invalid bus name: \"0dbus-app-snap.bar\"")
+}
+
+func (s *DbusAppInterfaceSuite) TestGetBusNamesNameMissingDot(c *C) {
+	var mockSnapYaml = []byte(`name: dbus-app-snap
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - dbus-app-snap
+`)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, Not(IsNil))
+	c.Assert(err, ErrorMatches, "invalid bus name: \"dbus-app-snap\"")
+}
+
+func (s *DbusAppInterfaceSuite) TestGetBusNamesNameMissingElement(c *C) {
+	var mockSnapYaml = []byte(`name: dbus-app-snap
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - dbus-app-snap.
+`)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, Not(IsNil))
+	c.Assert(err, ErrorMatches, "invalid bus name: \"dbus-app-snap\\.\"")
+}
+
+func (s *DbusAppInterfaceSuite) TestSanitizeSlotSystem(c *C) {
+	var mockSnapYaml = []byte(`name: dbus-app-snap
+version: 1.0
+slots:
+ dbus-app-slot:
+  interface: dbus-app
+  session:
+  - org.dbus-app-snap.system-1
+  - org.dbus-app-snap.system-2
+`)
+
+	info, err := snap.InfoFromSnapYaml(mockSnapYaml)
+	c.Assert(err, IsNil)
+
+	slot := &interfaces.Slot{SlotInfo: info.Slots["dbus-app-slot"]}
+	err = s.iface.SanitizeSlot(slot)
+	c.Assert(err, IsNil)
+}
+
+/*
```

Comment:
Oh, this was meant to be removed-- it's a test for the plug side of the connection which we don't support in this PR (perhaps in the future as per description in this PR).